File: Chapter_1/Chapter1_in_c++/two.cpp
/*  1.2
*   Write code to reverse a C-Style String.
*   (C-String means that “abcd” is represented as five characters, including the null character.)
*/
#include<bits/stdc++.h>
using namespace std;
string reverseString(string str)
{
   if(str.length() == 0)
   {
       return str;
   }
   int i = 0;
   int j = str.length() - 1;
   while(i < j)
   {
       char tmp = str[i];
       str[i] = str[j];
       str[j] = tmp;
       i++;
       j--;
   }
   return str;

}
int main()
{
    string input;
    cin>>input;
    cout<<reverseString(input)<<endl;
    return 0;
}

File: Chapter_1/Chapter1_in_Java/Two.java
package Chapter1;

import java.util.Scanner;

/*1.2
 * Write code to reverse a C-Style String.
 * (C-String means that “abcd” is represented as five characters, including the null character.)
 */

/**
 *
 * @author Dipty
 */
public class Two {
    String reverseString(String s)
    {
        int i = 0, j = s.length() - 1;
        if(s.isEmpty())
        {
            return s;
        }
        char[] str = s.toCharArray();
        while(i < j)
        {
            char tmp = str[i];
            str[i] = str[j];
            str[j] = tmp;
            i++;j--;
            
        }
        return new String(str);
    }
    public static void main(String[] args) {  
         String input;
        Scanner in = new Scanner(System.in);
        input = in.nextLine();
        Two t = new Two();
        System.out.println(t.reverseString(input));
    }
}
